File: ruby/BetaPowerEfficientSpaceStation.rb
# encoding: utf-8

require_relative 'Dice'
require_relative 'PowerEfficientSpaceStation'
require_relative 'BetaPowerEfficientSpaceStationToUI'

module Deepspace
    class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation
        @@EXTRAEFFICIENCY=1.2
        def initialize(station)
            super
            @dice = Dice.new
        end
    
        def fire
            if @dice.extraEfficiency
            return super*@@EXTRAEFFICIENCY
            else
            return super
            end
        end
    
        def getUIversion
            return BetaPowerEfficientSpaceStationToUI.new(self)
        end
    
        def to_s
            return "Beta" + super
        end
    end
end

File: java/Deepspace/src/deepspace/BetaPowerEfficientSpaceStation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author alvaro
 */
public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation{
    private Dice dice;
    private final float EXTRAEFFICIENCY = 1.2f;
    
    public BetaPowerEfficientSpaceStation(SpaceStation station){
        super(station);
        dice = new Dice();
    }
    
    @Override
    public float fire(){
        if(dice.extraEfficiency()){
            return super.fire()*EXTRAEFFICIENCY;
        }else return super.fire();
    }
    
    @Override
    public String toString(){
        return "BETA " + super.toString();
    }
    
    @Override
    public BetaPowerEfficientSpaceStationToUI getUIversion(){
        return new BetaPowerEfficientSpaceStationToUI(this);
    }
}
